### PR TITLE
Build with NPM

### PR DESCRIPTION
**Pull request by alexottoboni**
Builds Shdr with NPM. Thanks to @josepharhar.

### CODE REVIEW

File: build/build.js [hunk 1]
```diff
@@ -137,6 +138,8 @@ function main() {
 		}
 
 		var temp = buffer.join('');
+      recrmdirSync(getDirName(output));
+      fs.mkdirSync(getDirName(output));
```
Comment (BKcore) [inline build/build.js]: nit: match indentation

**Comment by BKcore** [comment]
Make sure you rebase on top of dev to get rid of the conflicts.

**Comment by BKcore** [comment]
Ok after looking into this a bit more, please move the package.json and .gitignore files to the top level directory since this is how modules should be structured. (make sure you modify npm start with the correct path).

After that this should be good to merge, thanks!

File: sources/shdr/App.coffee [hunk 2]
```diff
@@ -279,12 +279,12 @@ class App
       e.preventDefault?()
       false
     else if e.ctrlKey and e.altKey # Flip Shader
-        if @conf.mode is App.FRAGMENT
-            @setMode(App.VERTEX,true)
-            @ui.setMenuMode(App.VERTEX)
-        else
-            @setMode(App.FRAGMENT,true)
-            @ui.setMenuMode(App.FRAGMENT)
+      if @conf.mode is App.FRAGMENT
```
Comment (alexottoboni) [inline sources/shdr/App.coffee]: Indent isn't really related to NPM, but it wouldn't build without this change.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+build/node_modules
```
Comment (BKcore) [inline .gitignore]: this needs to be just `node_modules` now.

File: sources/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+shdr/*.js
+build/
```
Comment (BKcore) [inline sources/.gitignore]: This file is not needed anymore.

**Comment by BKcore** [comment]
Thanks!

**Pull request merged**